src/components/Datatable/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,206 @@
+/* eslint-disable react/boolean-prop-naming */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { getColor, pxToRem } from '../../../utils/helpers';
+import { FlexContainer } from '../../FlexContainer';
+import { SSCIconNames } from '../../../theme/icons/icons.enums';
+import { Spinner } from '../../Spinner';
+import { Label } from '../../forms/Label';
+import { PaginationProps } from './Pagination.types';
+import PageButton from './PageButton';
+import NavButton from './NavButton';
+import { StyledNumber } from '../../Filters/components/Number/Number';
+
+const MAX_PAGE_BUTTONS = 8;
+
+const PaginationWrapper = styled(FlexContainer)`
+  background-color: ${getColor('graphite3H')};
+  padding: ${pxToRem(16)};
+`;
+
+const Ellipsis = styled.div`
+  display: inline-block;
+  width: ${pxToRem(48)};
+  height: ${pxToRem(32)};
+  padding: ${pxToRem(8, 16)};
+  margin: ${pxToRem(0, 4)};
+  color: ${getColor('graphite2B')};
+  background-color: ${getColor('transparent')};
+`;
+
+const PageInputLabel = styled(Label)`
+  margin: ${pxToRem(6, 16)};
+`;
+const PageInput = styled(StyledNumber)`
+  width: ${pxToRem(72)};
+  height: ${pxToRem(32)};
+  padding: ${pxToRem(8, 16)};
+`;
+
+const Pagination: React.FC<PaginationProps> = ({
+  gotoPage,
+  previousPage,
+  nextPage,
+  pageCount,
+  canPreviousPage,
+  canNextPage,
+  pageIndex,
+  isLoading,
+}) => {
+  const pages = Array.from(Array(pageCount), (_, i) => i + 1);
+
+  const [visiblePageButtons, setVisiblePageButtons] = useState(pages);
+  const [isFirstEllipsisVisible, setIsFirstEllipsisVisible] = useState(false);
+  const [isLastEllipsisVisible, setIsLastEllipsisVisible] = useState(false);
+  const [isInputPageInvalid, setIsInputPageInvalid] = useState(false);
+
+  const onChangePageNumber = (e) => {
+    const pageNumber = Number(e.target.value);
+    if (pageNumber <= pageCount && pageNumber > 0) {
+      setIsInputPageInvalid(false);
+      gotoPage(pageNumber - 1);
+    } else {
+      setIsInputPageInvalid(true);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
I would at least add comments what is going on here and named the magic numbers, like `MID_SEQUENCE` etc. Also would worth to move the logic to custom hook then is easier to cover it by test. I still have a feeling it could be simplified 🙂

Reply:
This has to be simplified, I'd start with reverting logic of `if (pageCount > MAX_PAGE_BUTTONS)` we can do `if (pageCount <= MAX_PAGE_BUTTONS) return;` which remove one level of if nesting.

Also, it doesn't work, if I go to page 6 it shows only pages 1-6 and I can't go any further since it doesn't show page 7, same thing if I go from backwards on page 15 it show pages 15-20 and no page 14

Reply:
@PrincAm  @ajkl2533 There is a total of 4 cases as you can see in the Figma file ... working to simplify.
About the case of page 6, I agree with Radek, I will override the design and show 
[ 1 ...   5  **6**  7  8  ... N] same with 35 or N-5

Reply:
It's actually very intresting problem, I'm currently working on my own solution :D

Reply:
simplified in last commit